crates/consensus/src/pure/eip4844.rs

```diff
@@ -0,0 +1,150 @@
+use crate::constants::eip4844::{
+    BLOB_GASPRICE_UPDATE_FRACTION, BLOB_TX_MIN_BLOB_GASPRICE, TARGET_DATA_GAS_PER_BLOCK,
+};
+
+/// Calculates the `excess_blob_gas` from the parent header's `blob_gas_used` and `excess_blob_gas`.
+///
+/// See also [the EIP-4844 helpers](https://eips.ethereum.org/EIPS/eip-4844#helpers
+/// (`calc_excess_blob_gas`).
+#[inline]
+pub const fn calc_excess_blob_gas(parent_excess_blob_gas: u64, parent_blob_gas_used: u64) -> u64 {
```

Comment:
also cc @mattsse  I inhoused these 4844 functions which reth imports from revm in `reth/crates/primitives/src/eip4844.rs`. They were using identical constants under different names, so i've changed the names here. mind double-checking my work?

Reply:
this looks fine, but it makes me think that we need a crate for constants?
because maintaining this stuff twice is probably a bad idea?

we could add a lean alloy-eip-constants or something and then upstream to revm?

Reply:
it might make sense to do alloy-eips or something to house the pure logic, condtants, and helpers for each significant eip?

Reply:
sg!
should be very small, just constants+helper functions so we can use it in revm-primitives